Card guideline updates (#2467)
patternfly/patternfly-org: packages/v4/src/content/design-guidelines/demos/card-view/card-view.md

---
id: Card view
section: demos
---

A **card view** is a grid of cards in a gallery to facilitate browsing. Card views are typically used to present data set summaries, allowing users to drill down into any card to see more detailed content.

<img src="./img/card-view.png" alt="card view" />

Use a card view to:
- Display items in a data set that are best identified by a graphic or other visual representation.
- Make information easier to visualize and compare.
- Summarize a large number of objects at once within the same page.

> Do NOT use a card view to make a large amount of text-based content easier to scan. Use a table or list view instead.

Reviewer: Re-commenting to make sure Alana's previous suggestion doesn't get lost since it doesn't seem to have been addressed.

Two things, change "Do NOT" to "Never" and hyperlink "table" and "list view"!

![image](https://user-images.githubusercontent.com/51165119/117982528-1516f800-b304-11eb-9d5e-89f4a3b89bb7.png)

### Background colors for card views
Card views can be placed on a white or a gray background, depending on your use case.

**Use a white background** if your content view can change, such as when:
- A card that opens an expandable panel on the right.
- A card view that toggles into a table or list view.

> 

Reviewer: Adding a comment here to make sure Alana's previous comment/addition doesn't get lost.

She suggested adding "**Note**: If you use a white background, apply a gray border to your card instead of a drop shadow." 

![image](https://user-images.githubusercontent.com/51165119/117982361-edc02b00-b303-11eb-9991-8a0d9a010c31.png)

Author: I added a note with slightly different wording.

#### Inline actions
If an action can be performed on one card only, place the action in the card’s body or footer.

- If you have a few cards with single actions, use the [secondary button style](/components/button#variations).

- If you have many cards with single actions, use a [link button](/components/button#variations).

<img src="./img/few-multiple-actions.png" alt="few cards with multiple actions" />  

#### Multiple inline actions
If multiple actions can be performed on a single card, place those actions in a kebab. Position the kebab in the top-right corner of the card header.

<img src="./img/multiple-inline.png" alt="cards with multiple inline actions in a kebab" />

#### Global actions
If an action can be performed globally on multiple cards, or if it leads to multiple pieces of content at the same time, place the action in a toolbar above the card view. In the top-right corner of your card, add a checkbox so that users can select one or more items.

<img src="./img/global-actions.png" alt="card view with global action toolbar" />

#### Action card

> You may use an extra-small empty state inside a card to present users with another way to add more cards to the existing card view. We recommend using this feature **in addition to** a primary button in the toolbar, to ensure that the user is still able to perform the action from any page of the card view. Place the action card where a new card will be added, most likely in the very first or last card in the card view.

Reviewer: Should the last sentence here read "most likely **as** the very first or last card in the card view" instead of "**in** the very first"?

---
id: Card view
section: demos
---

A **card view** is a grid of cards in a gallery to facilitate browsing. Card views are typically used to present data set summaries, allowing users to drill down into any card to see more detailed content.

<img src="./img/card-view.png" alt="card view" />

Use a card view to:
- Display items in a data set that are best identified by a graphic or other visual representation.
- Make information easier to visualize and compare.
- Summarize a large number of objects at once within the same page.

Do NOT use a card view to make a large amount of text-based content easier to scan. Use a table or list view instead.

### Background colors for card views
Card views can be placed on a white or a gray background, depending on your use case.

> **Use a white backgroun**d if your content view can change, such as when:

Reviewer: the d at the end of background was left out of the bolding haha

<img src="./img/gray-background.png" alt="card view with gray background" />

### Card views on mobile
Cards will size down appropriately based on the built-in CSS responsiveness behaviors. Most of the time, they’ll follow the grid layout, expanding and compressing at certain breakpoints. At the smallest mobile size, they take up almost the entire width of the screen.

<img src="./img/mobile-cards.png" alt="mobile card view" width="375"/>

### Card view actions
You can customize card actions in a few ways. Depending on your use case, consider customizing inline actions, multiple inline actions, global actions, or action cards within your card view.

#### Inline actions
If an action can be performed on one card only, place the action in the card’s body or footer. You don’t need to include a checkbox because the actions are card-specific.

- If you have a few cards with single actions, use the [secondary button style](/components/button#variations).
- If you have multiple cards with single actions, use a [link button](/components/button#variations).

<img src="./img/few-multiple-actions.png" alt="few cards with multiple actions" />  

> Inline action cards are commonly used as navigation or drill-down indicators. In this case, present the user action on the card to differentiate this type of navigational card from a standard card. This way, users can easily distinguish the card types if your application includes both.   

Reviewer: I would maybe add "In this case, present the user action on the card **as opposed to in the toolbar**" (assuming that's what this means) just to clarify a little bit. Especially since at the end it says "if your application includes both" but it's not clear what "both" are? Inline action cards and what?

Author: I couldn't understand what this was trying to say, so I just removed the entire paragraph. Seemed more confusing than useful.

#### Multiple inline actions
If multiple actions can be performed on a single card, place those actions in a kebab. Position the kebab in the top-right corner of the card header.

<img src="./img/multiple-inline.png" alt="cards with multiple inline actions in a kebab" />

#### Global actions
If an action can be performed globally on multiple cards, or if it leads to multiple pieces of content at the same time, place the action in a toolbar above the card view. In the top-right corner of your card, add a checkbox so that users can select one or more items.

<img src="./img/global-actions.png" alt="card view with global action toolbar" />

#### Action card
You may use an extra-small empty state inside a card to present users with another way to add more cards to the existing card view. We recommend using this feature **in addition to** a primary button in the toolbar, to ensure that the user is still able to perform the action from any page of the card view. Place the action card where a new card will be added, most likely in the very first or last card in the card view.

<img src="./img/add-card-empty-state.png" alt="card view with addition card" />

### Images

> Only include meaningful images on your cards (don’t add them for decoration). Images should help users identify specific content types. Place smaller images above the card’s header, and place larger images within the card’s body.

Reviewer: I'm realizing now that maybe the bullet point in the github to add another image example here was referring to adding an image to show what placing "larger images within the card's body" would look like...

---
id: Card view
section: demos
---

A **card view** is a grid of cards in a gallery to facilitate browsing. Card views are typically used to present a summary of some data set where the user can drill-down from any card to see detailed content.

<img src="./img/card-view.png" alt="card view" />

Use a card view to:
- Display items in a data set that are best identified by a graphic or other visual representation.
- Make information easier to visualize and compare.
- Summarize a large number of objects at once within the same page.

> Do NOT use a card view to make a large amount of text-based content easier to scan. Use a table or list view instead.

Reviewer: This change doesn't seem to have been committed? Also, would link to table and list view guidelines here!

> Do NOT use a card view to make a large amount of text-based content easier to scan. Use a table or list view instead.

Reviewer: @mcarrano this still doesn't appear to have gone through?

### Background colors for card views

> Use a white or a gray background for your card views.

Reviewer 2: ```suggestion
Card views can feature a white or gray background, depending on your use case.
```

Use a white background if your content view can change. Examples:
- A card that opens an expandable panel on the right
- A card view that toggles into a table or list view

> 

Reviewer 2: ```suggestion
**Note**: If you use a white background, apply a gray border to your card instead of a drop shadow.
```

Bumping this note to above the example image, that way it won't get lost between the image and the next background section.

<img src="./img/white-background.png" alt="card view with white background" /> 

> **Note**: If you use a white background, apply a gray border to your card instead of a drop shadow.

Reviewer 2: ```suggestion
```

> Use a gray background if the sole purpose of your content view is to display cards.

Reviewer 2: ```suggestion
**Use a gray background** if the sole purpose of your content view is to display cards.
```

> <img src="./img/gray-background.png" alt="card view with gray background" />

Reviewer 2: In this image: Let's change "Cards" to lowercase in the page description and capitalize the "F" in "Patternfly" on the right.

### Card views on mobile
Cards will size down appropriately based on the built-in CSS responsiveness behaviors. Most of the time, they’ll follow the grid layout, expanding and compressing at certain breakpoints. At the smallest mobile size, they take up almost the entire width of the screen.

<img src="./img/mobile-cards.png" alt="mobile card view" width="375"/>

### Card view actions

> You can customize card actions in a few ways. Here are some recommendations to consider depending on your use case:

Reviewer 2: ```suggestion
You can customize card actions in a few ways. Depending on your use case, consider customizing inline actions, multiple inline actions, global actions, or action cards within your card view.
```

#### Inline actions
If an action can be performed on one card only, place the action in the card’s body or footer. You don’t need to include a checkbox because the actions are card-specific.

> - If you have a few cards with single actions, use the secondary button style.

Reviewer 2: Can we link to this button style?

> - If you have multiple cards with single actions, use a link button or link text.

Reviewer 2: Can we link to this button style/text style?

> <img src="./img/few-multiple-actions.png" alt="few cards with multiple actions" />  

Reviewer 2: Can we write "Cards" in lowercase in this image's page description?

Inline action cards are commonly used as navigation or drill-down indicators. In this case, present the user action on the card to differentiate this type of navigational card from a standard card. This way, users can easily distinguish the card types if your application includes both.   

<img src="./img/inline-actions.png" alt="cards with inline actions" />

#### Multiple inline actions

> If multiple actions can be performed on a single card, place those actions in a kebab. Position the kebab in the top-right corner of the header.

Reviewer 2: ```suggestion
If multiple actions can be performed on a single card, place those actions in a kebab. Position the kebab in the top-right corner of the card header.
```

> <img src="./img/multiple-inline.png" alt="cards with multiple inline actions in a kebab" />

Reviewer 2: Same small capitalization change (to lowercase) for "Cards" in this image's page description!

#### Global actions
If an action can be performed globally on multiple cards, or if it leads to multiple pieces of content at the same time, place the action in a toolbar above the card view. In the top-right corner of your card, add a checkbox so that users can select one or more items.

> 

Reviewer 2: ```suggestion

To perform a global action on one or more selected cards, users can check their desired cards' checkbox. Upon hover, a card's shadow expands, and their checkbox turns blue when selected. Once a user makes their desired card selections, they can perform global actions by clicking the global action button in the toolbar.
```

This might be made clearer by writing out the sequence vs. providing a general list of behavior. At first, I thought the numbered list indicated a procedure: But if what I'm gathering is correct, we're looking to summarize expected behavior here. Above is just a suggestion for how we can accomplish communicating the action sequence and expected behavior.

Let's also move this above the example image.

Author: I just removed this. It's not necessary.

> <img src="./img/global-actions.png" alt="card view with global action toolbar" />

Reviewer 2: Same small capitalization change for "Cards" in this example!

> 1. Upon hover, the card’s shadow expands.
> 2. Global actions are placed in the toolbar.
> 3. The card’s checkbox allows a user to perform an action on multiple cards. When a user selects a card, the checkbox turns blue to indicate selection.

Reviewer 2: ```suggestion
```

#### Action card

> You may use an extra-small empty state inside of a card to present the user with another means of adding a new card/item to their card view. If used, we recommend using it **in addition to** a primary button in a toolbar, to ensure that the user is still able to perform the action no matter what page of the card view they are on. The placement of the action card should be consistent with where a new card will be added, most likely the very first or the very last card in the card view.

Reviewer 2: ```suggestion
You may use an extra-small empty state inside a card to present users with another way to add more cards to the existing card view. We recommend using this feature **in addition to** a primary button in the toolbar, to ensure that the user is still able to perform the action from any page of the card view. Place the action card where a new card will be added, most likely in the very first or last card in the card view.
```

> <img src="./img/add-card-empty-state.png" alt="card view with addition card" />

Reviewer 2: Same small capitalization note: "Cards" to "cards" in the image's page description.

Author: I can't find the source for this image. @mmenestr any ideas?

### Images
Only include meaningful images on your cards (don’t add them for decoration). Images should help users identify specific content types. Place smaller images above the card’s header, and place larger images within the card’s body.

<img src="./img/card-images.png" alt="card view with logo images in each card" />

### Bottom pagination

> If your card view has multiple pages, add a footer with pagination.

Reviewer 2: Can we hyperlink to pagination here?

> <img src="./img/card-view-sticky-toolbar.png" alt="card view with sticky bottom toolbar" />

Reviewer 2: Same small capitalization note here: "Cards" to "cards" in the image's page description.

Author: The source for this image is missing, too!
